### PR TITLE
Remove the need for a separate branch in integration tests + cleanup

### PR DESCRIPTION
**Pull request by baldimir**
- Updates the dependencies versions. 
- Removes the need to have a separate branch to run integration tests correctly. 
- Adds proper README.

### CODE REVIEW

**Comment by baldimir** [comment]
@hellowdan please reassign your review request if needed.

File: README.md
Comment (jstastny-cz) [inline README.md]: Wouldn't it be more convenient to use the user property instead? Would definition of `maven.main.skip` property work the same? https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#skipMain
```
<properties>
  <maven.main.skip>true</maven.main.skip>
</properties>
```
Or at least add a note that the compiler config can be replaced by the property?

**Review by jstastny-cz — COMMENTED** [review]
LGTM, just a minor comment.

File: README.md
Comment (baldimir) [reply README.md]: Fixed.

**Review by hellowdan — APPROVED** [review]

**Comment by hellowdan** [comment]
I'm waiting for @MarianMacik review before merging it.

File: unpack-build-maven-plugin-itests/unpack-build-maven-plugin/pom.xml
Comment (MarianMacik) [inline unpack-build-maven-plugin-itests/unpack-build-maven-plugin/pom.xml:12]: Maybe add here an explanation why this has the same GAV as the "real" unpack-build-maven-plugin. Another option would be to rename this to `-dummy` and build it before the `tests` module and set up a clean plugin execution to clean it afterwards. Then the `tests` module would run.

File: unpack-build-maven-plugin-itests/tests/src/test/java/org/kie/unpackbuildplugin/integrationtests/UnpackBuildIT.java
Comment (MarianMacik) [inline unpack-build-maven-plugin-itests/tests/src/test/java/org/kie/unpackbuildplugin/integrationtests/UnpackBuildIT.java:34]:
```suggestion
```

File: pom.xml
Comment (MarianMacik) [inline pom.xml:110]: No longer needed?

File: README.md
Comment (MarianMacik) [inline README.md:41]: Maybe also the invoker option could be added.

**Comment by baldimir** [comment]
Sorry, I am closing this as it is not needed anymore. I found a more standard solution for the unpack, so maybe this plugin is not needed anymore. See e.g. here https://github.com/kiegroup/drools/blob/main/build-parent/pom.xml#L2483

**Pull request closed without merging**
